packages/esbuild-plugin/src/index.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ *
+ */
+
+const babel = require('@babel/core');
+const stylexBabelPlugin = require('@stylexjs/babel-plugin');
+const flowSyntaxPlugin = require('@babel/plugin-syntax-flow');
+const typescriptSyntaxPlugin = require('@babel/plugin-syntax-typescript');
+const jsxSyntaxPlugin = require('@babel/plugin-syntax-jsx');
+const path = require('path');
+const fs = require('fs/promises');
+
+const PACKAGE_NAME = 'esbuild-plugin-stylex';
+
+const IS_DEV_ENV =
+  process.env.NODE_ENV === 'development' ||
+  process.env.BABEL_ENV === 'development';
+
+function stylexPlugin({
+  dev = IS_DEV_ENV,
+  unstable_moduleResolution = { type: 'commonJS', rootDir: process.cwd() },
+  stylexImports = ['@stylexjs/stylex'],
+  // absolute path to bundled CSS file
+  absoluteFilePath = path.resolve(__dirname, 'stylex.css'),
+  babelConfig: { plugins = [], presets = [] } = {},
+  useCSSLayers = false,
+  ...options
+} = {}) {
+  return {
+    name: PACKAGE_NAME,
+    async setup({ onLoad, onEnd, initialOptions }) {
+      const stylexRules = {};
+
+      onEnd(async ({ outputFiles }) => {
+        const rules = Object.values(stylexRules).flat();
+
+        if (rules.length === 0) {
+          return;
+        }
+
+        const collectedCSS = stylexBabelPlugin.processStylexRules(
+          rules,
+          useCSSLayers,
+        );
+        const shouldWriteToDisk =
+          initialOptions.write === undefined || initialOptions.write;
+
+        if (shouldWriteToDisk) {
+          await fs.mkdir(path.dirname(absoluteFilePath), { recursive: true });
+          await fs.writeFile(absoluteFilePath, collectedCSS, 'utf8');
+
+          return;
+        }
+
+        outputFiles.push({
+          path: '<stdout>',
+          contents: new TextEncoder().encode(collectedCSS),
+          get text() {
+            return collectedCSS;
+          },
+        });
+      });
+
+      onLoad({ filter: /\.[jt]sx?$/ }, async (args) => {
```

Comment:
```suggestion
      onLoad({ filter: /\.[mc]?[jt]sx?$/ }, async (args) => {
```

To account for `.mjs` and `cjs` etc.